### PR TITLE
un-mute ActivateWatchTests, bump up logging, and remove explicit sleeps

### PR DESCRIPTION
**Pull request by jakelandis**
Address test failure on #30699

If (when?) this test fails again please obtain the following information:
* Copy of the relevant failure
* Copy of the reproduce line
* The Jenkins build link
* The Gradle scan link
* The relevant cluster logs from "Google Cloud Storage Upload Report" (link found in Jenkins build)

### CODE REVIEW

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java [hunk 1]
```diff
@@ -107,11 +103,11 @@ public void testDeactivateAndActivate() throws Exception {
         assertThat(getWatchResponse, notNullValue());
         assertThat(getWatchResponse.getStatus().state().isActive(), is(true));
 
-        logger.info("Sleeping for another five seconds, ensuring that watch is executed");
-        Thread.sleep(5000);
         refresh();
-        long count3 = docCount(".watcher-history*", matchAllQuery());
-        assertThat(count3, greaterThan(count1));
+        assertBusy(() -> {
```
Comment (hub-cap) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java]: I noticed you added the `5s` timeout to the first `awaitBusy()`, any reason it was not added here?

**Review by hub-cap — APPROVED** [review]
lgtm, 1 question

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java]: there is a subtle difference...

the call above on line 93 is awaitBusy, which in the happy case always take 5 seconds, so this test will always take at least 5 seconds (ensuring nothing changed in that time period).

the call just above on line 107 is assertBusy, which in the happiest case will not need any waiting at all, but can wait up to 10 seconds if needed. It's a subtle change from the min time the test will take to minor increase in the max time the test _can_ take.

Comment (hub-cap) [reply x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java]: yeesh, i didnt even see that. thanks for the explain.

**Comment by jpountz** [comment]
@jakelandis Is there anything left to backport?

**Comment by jakelandis** [comment]
I am trying to stabilize these tests in master before backporting. I will remove the backport versions and track this via the meta issue #42409

**Pull request merged**